streamalert/shared/config.py

```diff
@@ -242,32 +240,18 @@ def _validate_config(config):
     if TopLevelConfigKeys.LOGS in config:
         for log, attrs in config[TopLevelConfigKeys.LOGS].items():
             if 'schema' not in attrs:
-                raise ConfigError('The \'schema\' is missing for {}'.format(log))
+                raise ConfigError("The 'schema' is missing for {}".format(log))
 
             if 'parser' not in attrs:
-                raise ConfigError('The \'parser\' is missing for {}'.format(log))
+                raise ConfigError("The 'parser' is missing for {}".format(log))
 
     # Check if the defined sources are supported and report any invalid entries
-    if TopLevelConfigKeys.SOURCES in config:
-        supported_sources = {'kinesis', 's3', 'sns', 'stream_alert_app'}
-        if not set(config[TopLevelConfigKeys.SOURCES]).issubset(supported_sources):
-            missing_sources = supported_sources - set(config[TopLevelConfigKeys.SOURCES])
-            raise ConfigError(
-                'The \'sources.json\' file contains invalid source entries: {}. '
-                'The following sources are supported: {}'.format(
-                    ', '.join('\'{}\''.format(source) for source in missing_sources),
-                    ', '.join('\'{}\''.format(source) for source in supported_sources)
-                )
-            )
-
-        # Iterate over each defined source and make sure the required subkeys exist
-        for attrs in list(config[TopLevelConfigKeys.SOURCES].values()):
-            for entity, entity_attrs in attrs.items():
-                if TopLevelConfigKeys.LOGS not in entity_attrs:
-                    raise ConfigError('Missing \'logs\' key for entity: {}'.format(entity))
-
-                if not entity_attrs[TopLevelConfigKeys.LOGS]:
-                    raise ConfigError('List of \'logs\' is empty for entity: {}'.format(entity))
+    if TopLevelConfigKeys.CLUSTERS in config:
+        # Used to track duplicate sources in separate cluster config files
+        sources = set()
+        for cluster_name, cluster_attrs in config[TopLevelConfigKeys.CLUSTERS].items():
+            if 'data_sources' in cluster_attrs:
```

Comment:
what about `data_sources` not existing within the cluster config? should this raise a `ConfigError` exception? or log a warning?